### DISCUSSION TITLE
Gitea Support

### DISCUSSION THREAD

**Discussion by trymeouteh** [pacstall/pacstall#548] [Ideas]
Add the ability to grab software and to be able to self host a repository on a gitea instance.

Gitea is a open source, self hostable Github essentially.

**Comment by Elsie19** [comment]
This will be added in 2.0.0

**Reply by trymeouteh** [reply]
Is it now possible to easily create pacstall scripts for packages hosted on any gitea instance in 2.0.0?

**Reply by Elsie19** [reply]
This has been a feature (adding remote repos) for a long time. If you want to add a custom repo, you should use this syntax:
```
https://gitea.angry.im/PeterCxy/Shelter/raw/branch/master/
```

I just found some random repository on a random Gitea instance.

**Reply by D-Brox** [reply]
Basically, you can already add a repo hosted on a gitea instance using:
```bash
pacstall -A ${gitea_url}/raw/branch/master
```
Your repo just needs to follow the structure described [here](https://github.com/pacstall/pacstall-programs).

Anyway, Pacstall 2.0.0 also introduced locally hosted repos, using the absolute path to it.

**Comment by D-Brox** [comment]
We currently already support adding self-hosted repos.

Pacstall only github and gitlab repos are automatically parsed by pacstall to get the root for the raw files, but you can still add the root of the url to the raw files to your repolist. I'm not sure what that path would be with gittea, but you probably could use it without much trouble.
